src/resources/classes/asset.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 Cognite AS
+
+import { CogniteClient } from '../..';
+import * as types from '../../types/types';
+import { AssetList } from './assetList';
+
+export interface SubtreeOptions {
+  depth?: number;
+}
+export class Asset implements types.Asset {
+  public id: types.CogniteInternalId;
+  public externalId?: types.CogniteExternalId;
+  public parentId?: types.CogniteInternalId;
+  public name: types.AssetName;
+  public description?: types.AssetDescription;
+  public metadata?: types.Metadata;
+  public source?: types.AssetSource;
+  public lastUpdatedTime: Date;
+  public createdTime: Date;
+  public path: number[];
+  public depth: number;
+  private client: CogniteClient;
+
+  constructor(client: CogniteClient, props: types.Asset) {
+    this.client = client;
+    this.id = props.id;
+    this.externalId = props.externalId;
+    this.parentId = props.parentId;
+    this.name = props.name;
+    this.description = props.description;
+    this.metadata = props.metadata;
+    this.source = props.source;
+    this.lastUpdatedTime = props.lastUpdatedTime;
+    this.createdTime = props.createdTime;
+    this.path = props.path;
+    this.depth = props.depth;
+  }
+
+  public delete = () => {
```

Comment:
We should provide the same `recursive` options provided in the v1-branch

Reply:
Solved